.github/workflows/spelling.yml

```diff
@@ -0,0 +1,24 @@
+name: Spell checking
+on:
+  push:
+    branches:
+      - "**"
+    tags-ignore:
+      - "**"
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    - cron: '15 * * * *'
+
+jobs:
+  build:
+    name: Spell checking
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2.0.0
+      with:
+        fetch-depth: 5
+    - uses: check-spelling/check-spelling@0.0.16-alpha
```

Comment:
>check-spelling/check-spelling@0.0.16-alpha [](start = 12, length = 42)

I'm not familiar with GitHub Actions.

1. Will we need to update this version manually to keep up with new releases?
2. Do we need to do something to manage the secrets.GITHUB_TOKEN?  Or is that automatically done by the Actions infrastructure?
3. Generally, are there any other actions we need to take to manage this (beyond excludes, patterns, etc. for new changes)?

Thanks for your work here.

Reply:
1. In theory @dependabot can tell you about updates, but for the near term, I'll probably make the PR myself.
1. The `GITHUB_TOKEN` is automatic https://docs.github.com/en/free-pro-team@latest/actions/reference/authentication-in-a-workflow#about-the-github_token-secret
1. Beyond that and trying to avoid merging things with spelling errors (if the main branch is unhappy, anyone who branches from that point will be unhappy / confused), no.
   * There is the option to add an [advice.txt](https://github.com/microsoft/terminal/blob/main/.github/actions/spell-check/advice.txt) file which is included in the message the bot leaves, but that's totally tunable and that's really something you'll work through as a project.
   * Technically GitHub has a [budget](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions) for actions, it's theoretically possible for someone to run into this limit, I suspect that Microsoft has a reasonably large budget and you won't notice (certainly microsoft/terminal hasn't complained) -- if someone publicly runs into this, I'd love to hear about it. (Generally other actions are more expensive than this one -- which runs at about 1 minute and is thus billed as 1-2 units.) Most likely if a project like yours hits something like this, it means you have a bunch of files that should be ignored that you aren't ignoring (one project I'm looking at takes ~20 minutes to scan). -- If this becomes a problem for some other project in the future, I'll probably write some code so that it tries to scan [only changed files](https://github.com/check-spelling/check-spelling/wiki/Feature:-Only-changes) instead of the whole repository.

Reply:
Thanks for the info, I agree that only checking changed files seems like a good optimization opportunity.